muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/yo.tsx

```diff
@@ -1,64 +1,68 @@
 import * as React from 'react';
-import {connect, Dispatch} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import { connect, Dispatch } from 'react-redux';
+import { bindActionCreators } from 'redux';
 import * as queryString from 'query-string';
-import {i18nType} from '~/reducers/base/i18n';
+import { i18nType } from '~/reducers/base/i18n';
 import { RecordsType } from '~/reducers/main-function/records/records';
 import BodyScrollKeeper from '~/components/general/body-scroll-keeper';
 import Link from '~/components/general/link';
 import { UserWithSchoolDataType } from '~/reducers/main-function/user-index';
-import {StateType} from '~/reducers';
+import { StateType } from '~/reducers';
 import { HOPSType } from "~/reducers/main-function/hops";
 import mApi from '~/lib/mApi';
 import MatriculationSubjectType from './matriculation-subjects/matriculation-subject-type';
-
 import '~/sass/elements/empty.scss';
 import '~/sass/elements/loaders.scss';
 import '~/sass/elements/course.scss';
 import '~/sass/elements/application-sub-panel.scss';
+import moment from '~/lib/moment';
 import '~/sass/elements/buttons.scss';
 
-
-
 interface YOProps {
   i18n: i18nType,
   records: RecordsType,
   hops: HOPSType
 }
 
+type EligibilityStatus = "NOT_ELIGIBLE" | "ELIGIBLE" | "ENROLLED";
+
+interface Eligibility {
+  status: EligibilityStatus,
+  coursesCompleted: Number,
+  coursesRequired: Number,
+  enrollmentDate: String,
+  examDate: String
+}
+
 interface YOState {
+  eligibility?: Eligibility,
+  err?: String
   matriculationSubjects: MatriculationSubjectType[],
   matriculationSubjectsLoaded: boolean
 }
 
-class YO extends React.Component<YOProps,YOState> {
-  constructor(props:YOProps){
+class YO extends React.Component<YOProps, YOState> {
+  constructor(props: YOProps) {
     super(props);
 
     this.state = {
       matriculationSubjects: [],
       matriculationSubjectsLoaded: false
     }
   }
-  
-  /**
-   * Finds a matriculation subject name by subject value
-   * 
-   * @param code matriculation subject code
-   * @returns subject name or empty string if not found 
-   */
-  getMatriculationSubjectNameByCode = (code: string): string => {
-    return this.props.i18n.text.get(`plugin.records.hops.matriculationSubject.${code}`);
-  }
 
-  /**
-   * Component did mount life-cycle method  
-   * 
-   * Reads available matriculation subjects from REST API
-   */
   componentDidMount() {
+    mApi().records.studentMatriculationEligibility
```

Comment:
Can we use async and '~/util/promisify'?

Reply:
Added async and promisify